Initial support for PNG textures

garrettjoecox opened the pull request:
Alright, so I've been attempting to get familiar with ZAPD today. This might be already known but we don't really need the PngOTRizer, this functionality is mostly built into ZAPD. ~I'm not super happy with the solution I have of also adding our OTR texture headers in the ZAPD code~ Nvm this, no OTR headers necessary

In this this PR you just need to add an image anywhere under `assets/` with the extension of `name.{format}.png` like `assets/gRandThing.ia16.png` and it will be encoded and added to the OTR, and useable in C with `ResourceMgr_LoadFileRaw("assets/gRandThing")`

garrettjoecox committed:
Initial support for PNG textures

garrettjoecox commented:
Welp, it appears the OTR headers were completely unnecessary and it just works ™️ without them. I'm perfectly happy with this now. :D

dcvz reviewed (approved):
Great work! Easy solution for custom textures ✨

dcvz commented:
We can probably use this method now to do the dpad icon -- which we currently ship a `.bin` for

PurpleHato commented:
Thank you for this Proxy, will definetly feels better than having my temporary hardcoded texture in there too!
small nitpick the texture for these two RAND and MQ should be under **assets/textures/title_static/** and not **assets/ship_of_harkinian/textures/**
So like that they will be in the proper folder with all the other file_choose texture

aMannus commented:
This is great! :D

garrettjoecox commented:
> We can probably use this method now to do the dpad icon -- which we currently ship a `.bin` for

Do we have the original PNG for this?

PurpleHato commented:
> > We can probably use this method now to do the dpad icon -- which we currently ship a `.bin` for
> 
> Do we have the original PNG for this?

The person who added the dpad probably, but I forgot who added it

garrettjoecox committed:
Minor tweaks

briaguya0 commented:
> Do we have the original PNG for this?

yes, it was Colbydude https://github.com/Colbydude/soh-mods/blob/main/dpad-shortcuts/doc/dpad.png

garrettjoecox committed:
Move custom PNG handler to OTRExporter process

garrettjoecox commented:
Alright, moved the PNG conversion process to our OTRExporter and added the dpad texture. This required slight changes on the mac & linux scripts as they only handled things in assets/ship_of_harkinian rather than assets/*, windows is still broken, looking into it now

garrettjoecox commented:
Issue with windows has been sorted. This has been verified to be working on mac, windows, and linux.

garrettjoecox commented on ZAPDTR/ZAPD/ImageBackend.cpp:
```diff
@@ -147,7 +147,7 @@ void ImageBackend::ReadPng(const char* filename)
 void ImageBackend::ReadPng(const fs::path& filename)
 {
-	ReadPng(filename.c_str());
+	ReadPng(filename.string().c_str());
```
Fix from @Jack-Walker, will be PR'd into upstream
